### ISSUE TITLE
Add 'taplo' to the CI workflow to check the formatting of Cargo.toml files

### ISSUE THREAD

**Issue by ebma** [pendulum-chain/spacewalk#557]
## Context
There is a tool called [taplo](https://taplo.tamasfe.dev/) that can check and format TOML files. Since different IDEs by default have different rules for formatting TOML files, we want to streamline the formatting by using this tool.

## TODO

This would ideally be done only after #525 to make sure that the precommit hook is fixed. 

- Add taplo to the automated build checks in the Github CI workflow. Only check for wrong formatting with `taplo fmt --check` but don't correct the formatting in the CI
- Add a configuration file. We can put it into `.config/taplo.toml` and draw inspiration from the one used for the polkadot-sdk [here](https://github.com/paritytech/polkadot-sdk/blob/087ea0352ec2a3a2f0dc5254fe0455ba429bc25f/.config/taplo.toml)
- Add instructions to install and use taplo to the README. 
- Add `taplo fmt` to the precommit hook and make it format all the committed files automatically

**Comment by ebma** [comment]
@pendulum-chain/product a nice-to-have feature for developer experience. Kind of blocked by #525.

**Comment by prayagd** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Pendulum-6408a0365cda59c57d39bee5/issues/pendulum-chain/spacewalk/557?planning-poker) with Zenhub @b-yap @ebma @gianfra-t @TorstenStueber
